asciimatics/widgets/dropdownlist.py

```diff
@@ -118,10 +130,13 @@ def update(self, frame_no):
         # the clever stuff when it has the focus.
         text = "" if self._line is None else self._options[self._line][0]
         (colour, attr, background) = self._pick_colours("field", selected=self._has_focus)
+        if self._fit:
+            width = min(max(map(lambda x:len(x[0]), self._options)) + 1, self.width - 3)
+        else:
+            width = self.width - 3
         self._frame.canvas.print_at(
-            "[{:{}}]".format(
-                _enforce_width(text, self.width - 2, self._frame.canvas.unicode_aware),
-                self.width - 2),
+            "[ {:{}}]".format(
```

Comment:
Any reason that you wanted the extra whitespace around the visible elements?  I've not got strong feelings about keeping it the old way, but do have a couple of concerns...

1) I always worry about adding extra space in the limited real estate of most terminals...  
2) Does this mean that the text looks like it moves when you use the pop-up window to select a new value?

I can live with (1) if you feel strongly that it is a better aesthetic, but we definitely need to address (2) if the pop-up isn't correctly aligned.